pkg/controller/replication_controller.go

```diff
@@ -257,14 +257,18 @@ func (rm *ReplicationManager) deletePod(obj interface{}) {
 		}
 	}
 	if rc := rm.getPodControllers(pod); rc != nil {
-		rm.expectations.DeletionObserved(rc)
+		rcKey, err := controllerKeyFunc(rc)
+		if err != nil {
+			glog.Errorf("Couldn't get key for replication controller %+v: %v", rc, err)
```

Comment:
Added continue, and fixed similar issues in the replication controller.

So I wanted to use the same Expectations class for both rc and dc, so I generalized Expectations to accept a string argument instead of an rc. I tried to use ObjectMeta instead, but it didn't work out, because KeyFunc/rcKeyFunc doesn't work on ObjectMeta, but only on API types.

I guess another option is to refactor the key function, and then modify the Expectations class to accept ObjectMetas instead of string keys. Is that a better option?